[Mellanox]Correct reboot cause when reboot via power cycle

stephenxs opened the pull request:
Ignore reset_sw_reset which indicates rebooted by software.

**- What I did**
Correct reboot cause when reboot via power cycle by ignoring reset_sw_reset.

process-reboot-cause has the following logic:
1. check whether there is a hardware reboot cause, if so treats it as the final reboot-cause
2. otherwise it checks the previously saved reboot cause files in /host/reboot-cause/

Currently we implement "reboot" command via using power cycle. After a power-cycle reboot the reset_sw_reset contains a "1". Originally chassis reboot-cause handling treats it as a hardware-caused reboot and returns it to process-reboot-cause. In this case, the process-reboot-cause treats it as the reboot cause which is wrong.

To resolve that, just ignore reset_sw_reset.

**- How I did it**
Ignore reset_sw_reset

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

Commit:
[sonic_platform]remove the handling of reset_sw_reset which indicates rebooted by software.

keboliu reviewed (commented):
suggest not to ignore it but treat it as "none hardware"

Commit:
[sonic_platform]Check "reset_sw_reset"
Also check reboot cause file "reset_sw_reset" which indicates the system was rebooted due to software requesting.

stephenxs commented:
> suggest not to ignore it but treat it as "none hardware"

Fixed.

stephenxs commented:
Community PR opened.
https://github.com/Azure/sonic-buildimage/pull/3597
